### ISSUE TITLE
[ FEATURE REQUEST] Please add keyboard shortcuts

### ISSUE THREAD

**Issue by KruyKaze** [hzrd149/nostrudel#30]
Mainly:
j and k for next/previous post
l for "like"

Thanks for  Nostrudel

**Comment by weex** [comment]
I'd vote for "i" for new post as well and maybe "n" for notifications, "s" for search

**Comment by psic4t** [comment]
Yes, please! Im in. Really good shortcuts are in Pinafore (a Mastodon client) btw.

**Comment by hzrd149** [comment]
I really like this idea. there is probably a million different keyboard shortcut I could add but I think ill start with "j" and "k" for navigating the timeline. then a few for some of the common nav items.

However I cant promise that the keyboard shortcuts will work on all views. so you wont be able to navigate around the whole app with just your keyboard. but it should make browsing the timeline a little easier :)

**Comment by KruyKaze** [comment]
> I really like this idea. there is probably a million different keyboard shortcut I could add but I think ill start with "j" and "k" for navigating the timeline. then a few for some of the common nav items.
> 
> However I cant promise that the keyboard shortcuts will work on all views. so you wont be able to navigate around the whole app with just your keyboard. but it should make browsing the timeline a little easier :)

That would be a great start.  It's mostly needed in the timeline and when you click on a post to navigate between comments.

**Comment by hzrd149** [comment]
Added keyboard shortcuts to the navigation menu in the alpha version 

![Image](https://github.com/hzrd149/nostrudel/assets/8001706/56dbbae2-ab32-4f61-956a-c98279e7b965)

**Comment by hzrd149** [comment]
Released

**Comment by KruyKaze** [comment]
Hello @hzrd149 and thanks for your work on Nostrudel.
Any plans on implementing navigational keyboard shortcuts as in the initial request?

Mainly:
j and k for next/previous post
l for "like"

Thanks

**Comment by hzrd149** [comment]
hmm, I forgot about that. Ill re-open the issue so I hopefully wont forget again although ill admit it not very high on my priority list

**Comment by KruyKaze** [comment]
Great, thank you!

**Comment by badonyx** [comment]
I have some issues with the specific hotkeys implemented here.

- `⌘+L` interferes with browser hotkey for "Open Location", i.e. focus URL bar.
- `⌘+M` interferes with macos hotkey for minimize window.

**Comment by KruyKaze** [comment]
> I have some issues with the specific hotkeys implemented here.
> 
>     * `⌘+L` interferes with browser hotkey for "Open Location", i.e. focus URL bar.
> 
>     * `⌘+O` interferes with browser hotkey for "Open File..."
> 
>     * `⌘+M` interferes with macos hotkey for minimize window.
> 
> 
> Are they customizable? I could not find any settings for this.

Same here for ctrl+L on linux. Maybe these should be user set or toggleable

**Comment by Giszmo** [comment]
You forgot the most important of all keyboard shortcuts! I probably pressed ESC a solid twenty times already trying to back out from threads or viewing images.

**Comment by KruyKaze** [comment]
> You forgot the most important of all keyboard shortcuts! I probably pressed ESC a solid twenty times already trying to back out from threads or viewing images.

Yes that would be very helpful
